### DISCUSSION TITLE
Are BSim Postgres backend updates supported from the GUI?

### DISCUSSION THREAD

**Discussion by MatthewTingum** [NationalSecurityAgency/ghidra#6527] [Q&A]
From what I can tell, `CreateH2BSimDatabaseScript.java` and `AddProgramToH2BSimDatabaseScript.java` only support creating and updating local file-backed BSim databases from the GUI. Is this correct? If one wants to update a Postgres database backend they should use the `support/bsim` utility? As of writing, [BSim documentation](https://github.com/NationalSecurityAgency/ghidra/blob/c586600eabedc65293ee44143003211d4180977d/GhidraDocs/GhidraClass/BSim/BSimTutorial_Creating_Database_From_GUI.md) doesn't mention this.

**Comment by 01ChenQing** [comment]
I don't know the implementation details of Bsim, but I think Ghidra headless mode may be helpful to solve your problem.

**Comment by gemesa** [comment]
There are additional documents available under `Ghidra/Features/BSim/src/main/help/help/topics/BSim` including one specifically dedicated to the ingestion process: https://github.com/NationalSecurityAgency/ghidra/blob/c586600eabedc65293ee44143003211d4180977d/Ghidra/Features/BSim/src/main/help/help/topics/BSim/IngestProcess.html

Based on this only the `bsim` command can be used.
